From b6b0902044938b50b96ae9be60f01bc689ac306d Mon Sep 17 00:00:00 2001
From: Valentine Burley <valentine.burley@collabora.com>
Date: Mon, 7 Sep 2026 14:39:54 +0200
Subject: [PATCH] u_gralloc: Add and propagate dep_android_ui dependency

Add dep_android_ui to deps_for_u_gralloc and pass it to
idep_u_gralloc so that consumers linking against the u_gralloc
interface dependency properly inherit libui.

Signed-off-by: Valentine Burley <valentine.burley@collabora.com>
Reviewed-by: Yiwei Zhang <zzyiwei@chromium.org>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/44278>
---
 src/util/u_gralloc/meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/util/u_gralloc/meson.build b/src/util/u_gralloc/meson.build
index a4aee6f5c3f1..991e02f3c43d 100644
--- a/src/util/u_gralloc/meson.build
+++ b/src/util/u_gralloc/meson.build
@@ -22,6 +22,7 @@ if dep_android_ui.found()
   c_args_for_u_gralloc += '-DUSE_IMAPPER4_METADATA_API'
   cpp_args_for_u_gralloc += '-DUSE_IMAPPER4_METADATA_API'
   options_for_u_gralloc += 'cpp_std=c++17'
+  deps_for_u_gralloc += dep_android_ui
 elif dep_android_mapper4.found()
   files_u_gralloc += files('u_gralloc_imapper4_api.cpp')
   c_args_for_u_gralloc += '-DUSE_IMAPPER4_METADATA_API'
@@ -46,4 +47,7 @@ _libmesa_u_gralloc = static_library(
   dependencies: deps_for_u_gralloc,
 )
 
-idep_u_gralloc = declare_dependency(link_with: _libmesa_u_gralloc)
+idep_u_gralloc = declare_dependency(
+  link_with: _libmesa_u_gralloc,
+  dependencies: deps_for_u_gralloc,
+)
-- 
GitLab

